mir-ci/mir_ci/test_drag_and_drop.py

```diff
@@ -39,28 +55,39 @@ class TestDragAndDrop:
             ("python3", APP_PATH, "--source", "text", "--target", "text", "--expect", "text"),
         ],
     )
-    async def test_source_and_dest_match(self, modern_server, app) -> None:
-        modern_server = DisplayServer(modern_server, add_extensions=VirtualPointer.required_extensions)
-        program = modern_server.program(apps.App(app))
-
-        robot_test_case = f"""Source and Destination Match
-            Sleep     {STARTUP_TIME}
-            Move Pointer To Absolute    40    40
-            Sleep     {A_SHORT_TIME}
-            Press LEFT Button
-            Sleep     {A_SHORT_TIME}
-            Move Pointer To Absolute    120    70
-            Sleep     {A_SHORT_TIME}
-            Move Pointer To Absolute    200    100
-            Sleep     {A_SHORT_TIME}
-            Release LEFT Button
+    async def test_source_and_dest_match(self, modern_server, app, tmp_path) -> None:
+        extensions = VirtualPointer.required_extensions + ScreencopyTracker.required_extensions
+        server_instance = DisplayServer(modern_server, add_extensions=extensions)
+        program = server_instance.program(apps.App(app))
+
+        robot_settings = dedent(
+            f"""\
+            Library    {MIR_CI_PATH}/robot_libraries/WaylandHid.py
+            Resource   {MIR_CI_PATH}/robot_resources/screencopy.resource
+        """
+        ).strip("\n")
+
+        robot_test_case = dedent(
+            f"""\
+            Source and Destination Match
+                Sleep     {STARTUP_TIME}
+                ${{center}} =    Move Pointer Above Template    {MIR_CI_PATH}/robot_templates/drag_and_drop_src.png
+                Sleep     {A_SHORT_TIME}
+                Press LEFT Button
+                ${{off_center}} =    Displace    ${{center}}    20    20
+                Move Pointer To Absolute    ${{off_center}}[x]    ${{off_center}}[y]
```

Comment:
Why do we need this?

Reply:
This is from the original code; without it, `program.wait()` waits until the timeout.
The tests passes (locally) if I `program.kill()` it instead.

Reply:
Ah, I suppose this is because we're moving too far/quickly in one step. I suppose we'll want to interpolate the movements in the `Move Pointer` keyword in smaller increments to avoid this problem.

Reply:
Filed #72.